.github/workflows/version_check.yml

```diff
@@ -0,0 +1,82 @@
+name: Version Bump Check
+
+on:
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read
+
+jobs:
+  check-version-bump:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Use Node.js 18.x
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+
+      - name: Install dependencies
+        run: yarn
+
+      - name: Check for non-test file changes and version bump
+        run: |
+          # Get the base branch commit
+          BASE_SHA=${{ github.event.pull_request.base.sha }}
+          HEAD_SHA=${{ github.event.pull_request.head.sha }}
+
+          echo "Base SHA: $BASE_SHA"
+          echo "Head SHA: $HEAD_SHA"
+
+          # Get all changed files
+          CHANGED_FILES=$(git diff --name-only $BASE_SHA...$HEAD_SHA)
```

Comment:
This is a nitpick: we have a pattern of using the [tj-actions/changed-files job](https://github.com/transcend-io/privacy-types/blob/main/.github/workflows/package_version_bump.yml) to list out file changes as it removes all this boilerplate code, and is a well audited and maintained package. If we can, I'd recommend we switch the `git diff --name-only` portion to using that job, and perhaps even eliminate using Node entirely and use Bash scripts (but that's a tougher ask for sure)